hearth/lib/hearth/middleware/initialize.rb

```diff
@@ -0,0 +1,44 @@
+# frozen_string_literal: true
+
+module Hearth
+  module Middleware
+    # A middleware used to initialize the request, called first in the stack
+    # @api private
+    class Initialize
+      # @param [Class] app The next middleware in the stack.
+      # @param [List] interceptors
+      def initialize(app)
+        @app = app
+      end
+
+      # @param input
+      # @param context
+      # @return [Output]
+      def call(input, context)
+        # if there are exceptions, execution proceeds to before_completion hooks
+        before_execution_error = context.interceptors.apply(
+          hook: :read_before_execution, input: input,
+          context: context, output: nil, aggregate_errors: true
```

Comment:
nit: can we multiline all of these for readability?

Reply:
Was trying to avoid these methods getting too much longer.... but I can multiline.